ui/apps/platform/src/Containers/VulnMgmt/Reports/VulnMgmtReportForm.tsx

```diff
@@ -193,12 +193,15 @@ function VulnMgmtReportForm({
     }
 
     function onScheduledRepeatChange(_id, selection) {
-        // zero out the days selected list if changing interval type
-        if (selection !== values.schedule.intervalType) {
-            void setFieldValue('schedule.interval.days', []);
-        }
+        const intervalDayKey = selection === 'WEEKLY' ? 'daysOfWeek' : 'daysOfMonth';
+        const nextSchedule = {
```

Comment:
What do you think are pro and con of follow up to constrain the `Schedule` type so TypeScript compiler can verify the correctness of the conditional logic to build the `nextSchedule` object?

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/types/report.proto.ts#L39-L45

Reply:
Yeah absolutely. That is the route I started on but then backtracked in order to keep the change in this PR small. So I suppose the main con is that it would just be a larger change to review.

This section is tricky too, since the Formik `setFieldValue` API is `any`-typed, and most of the function calls throughout VM use implicit-any as well. I'd like to go through and get safer types written, but will likely wait until VM Report 2.0 to see how much code actually gets reused.